scenarios/common/runner.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+from functools import wraps
+
+import yaml
+
+from c7n import policy
+from c7n.config import Config
+from .base_terraform_test import get_resource_name
+
+output_dir = tempfile.mkdtemp()
+
+
+def policy_file(name, variables={}):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            cls = args[0]
+            with open(os.path.join(cls.policies_folder, name), 'r') as f:
+                data = yaml.load(f)

```

Comment:
why not just use policies.load ? the use case around trying to template this seems overly proscriptive a significant portion of action and filters do not have common representations, and those that that do typically have single common implementation.

Reply:
ie. this seems like a (de)/generative case of functional testing that is not really testing implementation, its testing permutation of usage which means its exponential growth in test cases and test time which is divorced of implementation testing, since its focused on expressibility testing. this seems like a secondary use case that should be layered on top of the a functional testing foundation for the subset of things to which it applies.